waspc/src/Wasp/Generator/WebAppGenerator/Vite.hs

```diff
@@ -0,0 +1,20 @@
+module Wasp.Generator.WebAppGenerator.Vite where
+
+import StrongPath (File', Path', Rel, relfile)
+import Wasp.AppSpec.ExternalCode (SourceExternalCodeDir)
+import qualified Wasp.AppSpec.ExternalCode as ExternalCode
+
+checkIfCustomViteConfigUsed :: [ExternalCode.File] -> Bool
```

Comment:
If we detect `vite.config.ts` or `vite.config.js` among the client external files: use the custom config to override the defaults.

Reply:
I would move this logic to `isCustomViteConfigUsed`, as suggested above.

Also, this logic has nothing to do with the Generator, since it is instead all about how our project is defined.